plugins/redfish/fu-self-test.c

```diff
@@ -87,6 +92,35 @@ fu_test_self_init(FuTest *self)
 		g_assert_no_error(error);
 		g_assert_true(ret);
 	}
+
+	/* HPE BMC */
+	self->hpe_plugin = fu_plugin_new_from_gtype(fu_redfish_plugin_get_type(), ctx);
+	ret = fu_plugin_runner_startup(self->hpe_plugin, progress, &error);
+	if (g_error_matches(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE)) {
+		g_test_skip("no redfish.py running");
+	} else {
+		g_assert_no_error(error);
+		g_assert_true(ret);
+
+		fu_redfish_plugin_set_credentials(self->hpe_plugin, "hpe_username", "password2");
+		/* We just changed the credentials and need to resetup again to discover the vendor
+		 * Here we need to get a device to query the backend
+		 */
```

Comment:
It doesn't need to be for this PR; but for the purpose of test coverage I think it would be a good idea to also have a negative test (IE send a bad user and then send a bad password)

Reply:
That would probably be a good idea "redfish wise", yes